frontend/src/metabase/components/Schedule/i18n.tsx

```diff
@@ -0,0 +1,117 @@
+import dayjs from "dayjs";
+import { isValidElement, cloneElement } from "react";
+
+import type { ScheduleDayType } from "metabase-types/api";
+
+import {
+  SelectWeekday,
+  SelectWeekdayOfMonth,
+  type SelectWeekdayProps,
+  SelectFrame,
+  type SelectFrameProps,
+  type SelectWeekdayOfMonthProps,
+} from "./components";
+import { getScheduleStrings } from "./constants";
+
+const isDay = (item: any, day: ScheduleDayType) =>
+  isValidElement(item) &&
+  (item.type === SelectWeekday || item.type === SelectWeekdayOfMonth) &&
+  (item as { props: SelectWeekdayProps }).props.schedule.schedule_day === day;
+
+const isSaturday = (item: any) => isDay(item, "sat");
+const isSunday = (item: any) => isDay(item, "sun");
+
```

Comment:
What follows is a small layer of code needed to correct a few mistranslations in relatively popular languages

These mistranslations are due to the composition of word translations into phrases, so they can't be straightforwardly remedied inside POEditor.

Reply:
We should fix the translations in POEditor and remove this code. I don't think really many folks on this team could maintain this and I don't think it's a pattern we'd want to follow for other features.